src/Glimpse.Server.Web/Storage/InMemoryStorage.cs

```diff
@@ -8,65 +7,200 @@ namespace Glimpse.Server.Web
 {
     public class InMemoryStorage : IStorage, IQueryRequests
     {
+        /// <summary>
+        /// Internal class to contain all data associated with a single request. 
+        /// </summary>
+        private class RequestInfo
+        {
+            public RequestInfo(LinkedListNode<Guid> requestIdNode)
+            {
+                this.RequestIdNode = requestIdNode;
+                this.Messages = new LinkedList<IMessage>();
+                this.Indices = null;
+            }
+
+            /// <summary>
+            /// List of individual messages associated with this request
+            /// </summary>
+            public LinkedList<IMessage> Messages { get; private set; }
+            
+            /// <summary>
+            /// The node in the _activeRequests list.  Storing this here allows us to determine to track 
+            /// requests by age in constant time.
+            /// </summary>
+            public LinkedListNode<Guid> RequestIdNode { get; private set; }
```

Comment:
Note that these are public methods on an private inner class, so external callers won’t have access.

From: Anthony van der Hoorn [mailto:notifications@github.com]
Sent: Friday, September 25, 2015 10:08 AM
To: Glimpse/Glimpse.Prototype <Glimpse.Prototype@noreply.github.com>
Cc: Mike Kaufman <Mike.Kaufman@microsoft.com>
Subject: Re: [Glimpse.Prototype] 55 in memory storage datastructures (#62)


In src/Glimpse.Server.Web/Storage/InMemoryStorage.cs<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fGlimpse%2fGlimpse.Prototype%2fpull%2f62%23discussion_r40452197&data=01%7c01%7cmike.kaufman%40microsoft.com%7c16035ee832c5419bea3908d2c5cbea6b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ZOCWSbtXOwfRvYwlfjwPC9JzL5eaX3JQf3rtcCb35qM%3d>:

> +            {

> +                this.RequestIdNode = requestIdNode;

> +                this.Messages = new LinkedList<IMessage>();

> +                this.Indices = null;

> +            }

> +

> +            /// <summary>

> +            /// List of individual messages associated with this request

> +            /// </summary>

> +            public LinkedList<IMessage> Messages { get; private set; }

> +

> +            /// <summary>

> +            /// The node in the _activeRequests list.  Storing this here allows us to determine to track

> +            /// requests by age in constant time.

> +            /// </summary>

> +            public LinkedListNode<Guid> RequestIdNode { get; private set; }

Is there a need for RequestIdNode and Messages to be public... given how this links into various indices, etc if someone mucked around with these strcutures directly, this could cause an issue. If we want to have a public list of these, maybe it should be an IEnumerable which lazily yeilds the result.

—
Reply to this email directly or view it on GitHub<https://na01.safelinks.protection.outlook.com/?url=https%3a%2f%2fgithub.com%2fGlimpse%2fGlimpse.Prototype%2fpull%2f62%2ffiles%23r40452197&data=01%7c01%7cmike.kaufman%40microsoft.com%7c16035ee832c5419bea3908d2c5cbea6b%7c72f988bf86f141af91ab2d7cd011db47%7c1&sdata=ACttfl3EqgnJkCYKxkSEGCw7CLGUi50USG7rgOaTSeQ%3d>.